engine/classes/Elgg/Filesystem/Adapter/AwsS3.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace Elgg\Filesystem\Adapter;
+use Gaufrette\Filesystem;
+use Gaufrette\File;
+use Gaufrette\Adapter\AwsS3 as GaufretteAwsS3;
+use Aws\S3\S3Client;
+
+class AwsS3 extends Filesystem implements Adapter {
+	private $config;
+	public function __construct(array $config) {
```

Comment:
I think the typical place for this now is in the dependency injector or some factory function. I'd much prefer to keep our constructors really dumb (field assignments only). If we do that I think we may even be able to drop another class in this case, too.